backend/backend.go

```diff
@@ -63,7 +63,6 @@ func connectToProxyWS(ws *websocket.Conn, handlers map[string]Handler) error {
 		_, msg, err := ws.ReadMessage()
 		if err != nil {
 			log.WithFields(log.Fields{"error": err}).Error("Received error reading from socket. Exiting.")
-			close(responseChannel)
```

Comment:
If you do this, will the running handler be stopped  or will it run forever and we''ll leak goroutines?
Maybe we need to call closeHandler() here?
Can you possibly test/debug this to figure that out?

Reply:
@cjellick i changed to close responders, and that should gracefully close handler we opened in go routines.